From 6160462beaf2d0308cd5a0556de3d17abc01cb77 Mon Sep 17 00:00:00 2001
From: Gerd Hoffmann <kraxel@redhat.com>
Date: Tue, 23 Sep 2014 14:56:08 -0400
Subject: [drm] bochs: Remove unnecessary NULL check in bo_unref

Message-id: <1411484172-28849-6-git-send-email-kraxel@redhat.com>
Patchwork-id: 96486
O-Subject: [RHEL-7.1 kernel PATCH 5/9] drm/bochs: Remove unnecessary NULL check in bo_unref
Bugzilla: 1043379
RH-Acked-by: Dave Airlie <airlied@redhat.com>

From: Daniel Vetter <daniel.vetter@ffwll.ch>

ttm_bo_unref unconditionally calls kref_put on it's argument, so the
thing can't be NULL without already causing Oopses.

Spotted by coverity.

Reviewed-by: David Herrmann <dh.herrmann@gmail.com>
Signed-off-by: Daniel Vetter <daniel.vetter@ffwll.ch>
(cherry picked from commit dcb1ee57780f2f772015ce9f80b2adc671257228)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/gpu/drm/bochs/bochs_mm.c b/drivers/gpu/drm/bochs/bochs_mm.c
index ce68587..ec48746 100644
--- a/drivers/gpu/drm/bochs/bochs_mm.c
+++ b/drivers/gpu/drm/bochs/bochs_mm.c
@@ -432,9 +432,7 @@ static void bochs_bo_unref(struct bochs_bo **bo)
 
  tbo = &((*bo)->bo);
  ttm_bo_unref(&tbo);
- if (tbo == NULL)
-  *bo = NULL;
-
+ *bo = NULL;
 }
 
 void bochs_gem_free_object(struct drm_gem_object *obj)
-- 
1.7.1